ci/gpu/build.sh

```diff
@@ -53,13 +53,13 @@ gpuci_conda_retry install -c conda-forge -c rapidsai -c rapidsai-nightly -c nvid
       "libcumlprims=${MINOR_VERSION}" \
       "dask-cudf=${MINOR_VERSION}" \
       "dask-cuda=${MINOR_VERSION}" \
-      "ucx-py=0.20.*" \
+      "ucx-py=0.21" \
       "ucx-proc=*=gpu" \
-      "xgboost=1.4.2dev.rapidsai${MINOR_VERSION}" \
       "rapids-build-env=${MINOR_VERSION}.*" \
       "rapids-notebook-env=${MINOR_VERSION}.*" \
       "rapids-doc-env=${MINOR_VERSION}.*" \
       "shap>=0.37,<=0.39"
+      # "xgboost=1.4.2dev.rapidsai${MINOR_VERSION}" \
```

Comment:
xgboost 1.4.2 packages for 21.08 have just been published. you should be okay to uncomment this line now.

https://anaconda.org/rapidsai-nightly/xgboost/files?version=1.4.2dev.rapidsai21.08

Reply:
yeah it was temporary just to test CI of 21.08

Reply:
@ajschmidt8 corrected this, could use your review/approval to merge